src/main/java/org/opensearch/knn/common/KNNVectorUtil.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.common;
+
+public class KNNVectorUtil {
+    private KNNVectorUtil() {}
+
+    /**
+     * Check if all the elements of a given vector are zero
+     *
+     * @param vector the vector
+     * @return true if yes; otherwise false
+     */
+    public static boolean isZeroVector(byte[] vector) {
+        for (byte e : vector) {
+            if (e != (byte) 0) {
```

Comment:
qq: Do we need 0 cast?

Reply:
not necessary, it's just my habit. I can remove it.